src/addons/mod/forum/pages/search/search.ts

```diff
@@ -0,0 +1,142 @@
+import { CoreSites } from '@services/sites';
+import { CoreDomUtils } from '@services/utils/dom';
+import { CoreUtils } from '@services/utils/utils';
+
+@Component({
+    selector: 'page-addon-mod-forum-search',
+    templateUrl: 'search.html',
+    styleUrls: ['search.scss'],
+})
+export class AddonModForumSearchPage implements OnInit {
+
+    loadMoreError: string | null = null;
+    searchBanner: string | null = null;
+    resultsSource = new CoreSearchGlobalSearchResultsSource('', {});
+    forum?: AddonModForumData;
+    searchAreaId?: string;
+
+    private ready = new CorePromisedValue<void>();
+
+    /**
+     * @inheritdoc
+     */
+    async ngOnInit(): Promise<void> {
+        try {
+            const site = CoreSites.getRequiredCurrentSite();
+            const searchBanner = site.config?.searchbanner?.trim() ?? '';
+            const courseId = CoreNavigator.getRequiredRouteNumberParam('courseId');
+            const forumId = CoreNavigator.getRouteNumberParam('forumId');
+            const filters: CoreSearchGlobalSearchFilters = {
+                courseIds: [courseId],
+            };
+
+            if (CoreUtils.isTrueOrOne(site.config?.searchbannerenable) && searchBanner.length > 0) {
+                this.searchBanner = searchBanner;
+            }
+
+            if (forumId) {
+                this.forum = await AddonModForum.getForumById(courseId, forumId);
+                const module = await CoreCourse.getModule(this.forum.cmid, courseId);
+
+                filters.searchAreaIds = ['mod_forum-post'];
+
+                if (module.contextid) {
+                    filters.contextIds = [module.contextid];
+                }
+
+                this.searchAreaId = `AddonModForumSearch-${courseId}-${this.forum.id}`;
+            } else {
+                filters.searchAreaIds = ['mod_forum-activity', 'mod_forum-post'];
+                this.searchAreaId = `AddonModForumSearch-${courseId}`;
+            }
+
+            this.resultsSource.setFilters(filters);
+            this.ready.resolve();
+        } catch (error) {
+            CoreDomUtils.showErrorModal(error);
+            CoreNavigator.back();
+
+            return;
+        }
+    }
+
+    /**
+     * Perform a new search.
+     *
+     * @param query Search query.
+     */
+    async search(query: string): Promise<void> {
+        await this.ready;
+
+        this.resultsSource.setQuery(query);
+
+        if (this.resultsSource.hasEmptyQuery()) {
+            return;
+        }
+
+        await CoreDomUtils.showOperationModals('core.searching', true, async () => {
+            await this.resultsSource.reload();
+            await CoreUtils.ignoreErrors(
+                CoreSearchGlobalSearch.logViewResults(this.resultsSource.getQuery(), this.resultsSource.getFilters()),
+            );
+        });
+    }
+
+    /**
+     * Clear search results.
+     */
+    clearSearch(): void {
+        this.loadMoreError = null;
```

Comment:
```suggestion
        this.loadMoreError = null;
        
        this.resultsSource.setQuery('');
        this.resultsSource.reset();
```
You should reset the source here too, right?